src/gui/accountsettings.cpp

```diff
@@ -96,6 +101,10 @@ bool showEnableE2eeWithVirtualFilesWarningDialog()
     encryptButton->setText(AccountSettings::tr("Encrypt folder"));
     e2eeWithVirtualFilesWarningMsgBox.exec();
 
+    SyncJournalFileRecord record;
+    if (!folder->journalDb()->getFileRecord(path, &record) || !record.isValid()) {
```

Comment:
@FlexW I think this check must be made after the dialog was closed.

Reply:
exec() returns if the dialog was closed.

Reply:
@FlexW Ok. But still, is it the correct place to check for folder existence? I mean, it's the code for the warning dialog, and, it will even be removed once we get the E2EE with VFS PR in, so this check will then be gone. Is it possible that the folder gets removed during us clicking the "Encrypt" even without having a warning dialog displayed? If you know what I mean.

Reply:
No, I think the check is only necessary (and I think even then it's very rare that a user might encounter that case) when the dialog is displayed. I wanted to have it close to the dialog because then it's easier to remove it once the dialog is removed.